### PR TITLE
add refresh token crud and authentication with middleware

### PR DESCRIPTION
**Pull request by SUSHILkhadka**

### CODE REVIEW

**Review by Panas18 — APPROVED** [review]
LGTM

**Review by beingbiplov — APPROVED** [review]
Looks good to me!

File: src/middlewares/authenticate.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { NextFunction, Response } from "express"
+import { StatusCodes } from "http-status-codes"
+import jwt  from "jsonwebtoken"
+import { IDataAtToken } from "../domains/IDataAtToken"
+import { IRequestWithTokenData } from "../domains/IRequestWithTokenData"
+import CustomError from "./CustomError"
+
+const authenticate = async (req:IRequestWithTokenData,res:Response,next: NextFunction)=>{
+    if(!req.headers.authorization){
+        // throw new CustomError("need token",StatusCodes.FORBIDDEN
```
Comment (aman007shrestha) [inline src/middlewares/authenticate.ts:10]: next(new CustomeError("need token", 500))

File: src/router/loginRouter.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Router } from 'express';
-import * as loginController from '../controllers/LoginController';
+import * as loginController from '../controllers/loginController';
 
 const router = Router();
 router.all('/', loginController.login);
```
Comment (aman007shrestha) [inline src/router/loginRouter.ts:5]: Why all? Use Post for login

File: src/router/registerRouter.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Router } from 'express';
+import * as userController from '../controllers/userController';
+
+const router = Router();
+router.all('/', userController.createUser);
```
Comment (aman007shrestha) [inline src/router/registerRouter.ts:5]: use post here as well.

**Review by aman007shrestha — COMMENTED** [review]
Kindly check comments

**Pull request merged**